### PR TITLE
Expand email validation for AB#16520

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes [AB#16520](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/16520)

## Description

1. Replace vuelidate's email validator with custom regex.
2. Add the new validation to registration and user profile email update.

## Testing

- [X] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

Unit Tests
image

Functional Tests
image

## UI Changes
image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/constants/validationRegEx.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ﻿export default abstract class ValidationRegEx {
     public static PhoneNumberMasked = /^\D?(\d{3})\D?\D?(\d{3})\D?(\d{4})$/;
+    public static Email =
+        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/constants/validationRegEx.ts:4]: Ah, yes, this looks right.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/GatewayApi/src/Validations/OptionalEmailAddressValidator.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    /// <summary>
+    /// Class encapsulating validation for email addresses.
+    /// </summary>
+    public class OptionalEmailAddressValidator : AbstractValidator<string>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OptionalEmailAddressValidator"/> class.
+        /// </summary>
+        public OptionalEmailAddressValidator()
```
Comment (BrianMaki) [inline Apps/GatewayApi/src/Validations/OptionalEmailAddressValidator.cs:29]: Is there a unit test for this?

**Review by BrianMaki — CHANGES_REQUESTED** [review]
For code coverage shouldn't there be a unit test for this?

File: Apps/GatewayApi/src/Validations/OptionalEmailAddressValidator.cs (hunk 2 above)
Comment (BronzBierd) [reply Apps/GatewayApi/src/Validations/OptionalEmailAddressValidator.cs:29]: good catch, I'll add it

**Review by sslaws — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**